src/Prima/Pyxis/ButtonGroup.elm

```diff
@@ -0,0 +1,299 @@
+    , classList : List ( String, Bool )
+    , attributes : List (Html.Attribute msg)
+    , id : Maybe String
+    }
+
+
+type ButtonGroupOption msg
+    = Class String
+    | ClassList (List ( String, Bool ))
+    | Attribute (Html.Attribute msg)
+    | Id String
+
+
+defaultOptions : Options msg
+defaultOptions =
+    { classes = []
+    , classList = []
+    , attributes = []
+    , id = Nothing
+    }
+
+
+{-| Internal. Applies the customizations made by end user to the `Button` component.
+-}
+applyOption : ButtonGroupOption msg -> Options msg -> Options msg
+applyOption modifier options =
+    case modifier of
+        Class class ->
+            { options | classes = class :: options.classes }
+
+        ClassList list ->
+            { options | classList = List.append list options.classList }
+
+        Attribute attr ->
+            { options | attributes = attr :: options.attributes }
+
+        Id id ->
+            { options | id = Just id }
+
+
+{-| Internal. Applies all the customizations and returns the internal `Options` type.
+-}
+computeOptions : Config msg -> Options msg
+computeOptions (Config config) =
+    List.foldl applyOption defaultOptions config.options
+
+
+{-| Internal. Adds a generic option to the `Button`.
+-}
+addOption : ButtonGroupOption msg -> Config msg -> Config msg
+addOption option (Config buttonGroupConfig) =
+    Config { buttonGroupConfig | options = buttonGroupConfig.options ++ [ option ] }
+
+
+{-| Create a button group with a `SpaceBetween` alignment.
+
+    --
+
+
+    import Prima.Pyxis.ButtonGroup as ButtonGroup
+
+    myBtnGroup : ButtonGroup.Config Msg
+    myBtnGroup =
+        ButtonGroup.spaceBetween myButtons
+
+-}
+spaceBetween : List (B.Config msg) -> Config msg
+spaceBetween buttons =
+    Config (ButtonGroupConfig buttons SpaceBetween [])
+
+
+{-| Create a button group with a `SpaceEvenly` alignment.
+-}
+spaceEvenly : List (B.Config msg) -> Config msg
+spaceEvenly buttons =
+    Config (ButtonGroupConfig buttons SpaceEvenly [])
+
+
+{-| Create a button group with a `SpaceAround` alignment.
+-}
+spaceAround : List (B.Config msg) -> Config msg
+spaceAround buttons =
+    Config (ButtonGroupConfig buttons SpaceAround [])
+
+
+{-| Create a button group with a `Centered` alignment.
+-}
+centered : List (B.Config msg) -> Config msg
+centered buttons =
+    Config (ButtonGroupConfig buttons Centered [])
+
+
+{-| Create a button group with a `ContentStart` alignment.
+-}
+contentStart : List (B.Config msg) -> Config msg
+contentStart buttons =
+    Config (ButtonGroupConfig buttons ContentStart [])
+
+
+{-| Create a button group with a `ContentEnd` alignment.
+-}
+contentEnd : List (B.Config msg) -> Config msg
+contentEnd buttons =
+    Config (ButtonGroupConfig buttons ContentEnd [])
+
+
+{-| Create a button group with a `CoverFluid` alignment.
+-}
+coverFluid : List (B.Config msg) -> Config msg
+coverFluid buttons =
+    Config (ButtonGroupConfig buttons CoverFluid [])
```

Comment:
Non sono molto d'accordo demandare al componente la predisposizione all'interno del suo spazio, secondo sono regole di stile che dovrebbero essere applicate da colui che invoca il componente

Reply:
Questo é per evitare proprio che sia l'utilizzatore a dover conoscere il dettaglio implementativo di come i bottoni sono organizzati. Considerato che il Group altro non fa che raggruppare, gli rimarrebbe poca utilità se non definire il come raggruppare i pulsanti al suo interno.

Reply:
mm dici? perche il dettaglio implementativo scusa? in realta l'utilizzatore predisponde graficamente i bottoni all'interno del suo spazio, non ha accesso al dettaglio implementativo

Infatti io sono un po scettico in generale sulla necessita di un btnGrp, pero come messo nel commento generale della pr, credo sia piu per gusto personale, visto che alcuni design system lo implementano e altri no.

Reply:
Possiamo fare una via di mezzo facendo in modo che `Alignment` non sia obbligatorio ma si possa aggiungere via `withAlignment` @peppescg @Zsavajji